lib/fshash/bucket.go

```diff
@@ -68,26 +46,18 @@ type Bucket interface {
 	question of which should be sorted first when creating a deterministic
 	hash of the whole tree.)
 */
-var FileCollision *errors.ErrorClass = errors.NewClass("FileCollision")
-
-var _ Bucket = &MemoryBucket{}
-
-type MemoryBucket struct {
-	// my kingdom for a red-black tree or other sane sorted map implementation
-	lines []line
-}
+var FileCollision *errors.ErrorClass = InvalidFilesystem.NewClass("FileCollision")
 
-type line struct {
-	metadata    Metadata
-	contentHash []byte
-}
+/*
+	MissingTree is reported when iteration over a filled bucket encounters
+	a file that has no parent nodes.  I.e., if there's a file path "./a/b",
+	and there's no entries for "./a", it's a MissingTree error.
+*/
+var MissingTree *errors.ErrorClass = InvalidFilesystem.NewClass("MissingTree")
 
-type linesByFilepath []line
+// for sorting
+type linesByFilepath []Record
```

Comment:
You can just refer to it as '[]Record`; that's already exported.  I wouldn't even make a type for this if that wasn't how one gets sorting behavior.

Also, nothing with type `[]Record` is ever returned, and here's why: the iterator interface is supposed to make sense for things bigger than memory so we're able to drop in a boltdb (or something) backed thing for handling absurdly large datasets where the metadata gets ponderous to handle in memory.  The iterator interface can hide that.  This detail of the memory bucket can't.